### PR TITLE
Feature/main/send emailjs service

### PR DESCRIPTION
**Pull request by jasc66**
# Description

Se empiezan pruebas con formulario de contacto para envio de correos, se configura y se pone a trabajar, sin embargo se debe optimizar el codigo y crear un ts para las funciones creadas. 

[To obtain the URL, navigate to the ticket on Trello and click on the "Share" button located at the bottom right corner.
Paste between the parentheses: [Trello Ticket URL]()](https://trello.com/c/XGsDR1hr/16-contact-form-email)

# Checklist
Please ensure that you have completed the following tasks:

- [ x] I have performed a self-review of my own code
- [ ] I have run ``` npm run lint ``` to automatically fix any linting issues
- [ ] I have resolved all linting issues within the files modified in this PR
- [x ] I have provided visual evidence of the changes (e.g., screenshots, GIFs, or videos)
- [ ] I have added comments to my code, particularly in difficult-to-understand areas
- [ ] I have made corresponding updates to the documentation, if necessary
- [ ] My changes generate no new warnings in the terminal or browser console

# Visual Evidence
image

image

# Test Coverage (Optional)
If applicable, please attach screenshots of the test coverage report:

Run npm run ``` test:coverage:changed ``` to generate the report
Note: Including test coverage is optional for now, but highly encouraged. (This is a task we will implement soon).

### CODE REVIEW

**Review by m-ret — APPROVED** [review]

File: components/Contact/index.tsx [hunk 1]
```diff
@@ -1,15 +1,61 @@
...
+      console.log('Form is not defined.');
+      return;
+    }
+
+    emailjs
+      .sendForm(
+        'service_q2yyuvc',
+        'template_ck5shih',
+        formRef,
+        'M6Cdv9mT9s7MHOoPM'
```
Comment (m-ret) [inline components/Contact/index.tsx:24]: Esta data debería de estar en un .env o en un ENUM.

**Review by m-ret — APPROVED** [review]

File: components/Contact/index.tsx (hunk 1 above)
Comment (jasc66) [reply components/Contact/index.tsx:24]: Se crean variables de entorno y se realizan pruebas de envio, de momento se tiene una alerta local que debe ser cambiada por un mensaje en el formulario 
image

File: components/Contact/index.tsx [hunk 2]
```diff
@@ -1,15 +1,68 @@
+'use client';
+
 import NewsLatterBox from './NewsLatterBox';
+import React, { useRef, useState, useEffect } from 'react';
+import emailjs from 'emailjs-com';
```
Comment (m-ret) [inline components/Contact/index.tsx:5]: Siempre los módulos internos creados por nosotros van debajo de los externos.
```js
import React, { useRef, useState, useEffect } from 'react';
import emailjs from 'emailjs-com';
import NewsLatterBox from './NewsLatterBox';
```

File: components/Contact/index.tsx [hunk 3]
```diff
@@ -1,15 +1,68 @@
...
+
+      .then(
+        result => {
+          console.log(result.text);
+          clearForm();
+          setShowAlert(true);
+        },
+        error => {
+          console.log(error.text);
+        }
```
Comment (m-ret) [inline components/Contact/index.tsx:35]: Aquí debería de haber un mejor manejo del error. Usemos un toast para eso: https://www.npmjs.com/package/react-toastify o este https://react-hot-toast.com/

Instale ese paquete, ya le hago un ticket.

**Comment by m-ret** [comment]
El ticket de las alertas: https://trello.com/c/ePlzGpsN/32-crear-un-sistema-de-alertas

File: components/Contact/index.tsx [hunk 4]
```diff
@@ -1,15 +1,68 @@
...
+
+    if (!formRef) {
+      console.log('Form is not defined.');
+      return;
+    }
+
+    emailjs
+      .sendForm(SERVICE_ID, TEMPLATE_ID, formRef, USER_ID)
+
+      .then(
```
[the remark this answers is not in the document]
Comment (jasc66) [reply components/Contact/index.tsx]: se elmina espacio

File: components/Contact/index.tsx (hunk 3 above)
Comment (jasc66) [reply components/Contact/index.tsx:34]: Se va implementar con el ticket

File: components/Contact/index.tsx [hunk 5]
```diff
@@ -1,15 +1,68 @@
...
+  const clearForm = () => {
+    const { current: formRef } = form;
+
+    if (!formRef) {
+      console.log('Form is not defined.');
+      return;
+    }
+    formRef.reset
+      ? formRef.reset()
+      : console.log('Reset function is not defined.');
```
[the remark this answers is not in the document]
Comment (jasc66) [reply components/Contact/index.tsx:46]: Es correcto solo limpia el formulario, prefiere que la modifique?

**Pull request merged**